titus-server-runtime/src/main/java/com/netflix/titus/runtime/connector/common/replicator/StreamDataReplicator.java

```diff
@@ -0,0 +1,94 @@
+package com.netflix.titus.runtime.connector.common.replicator;
+
+import java.util.concurrent.CountDownLatch;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.atomic.AtomicReference;
+import javax.annotation.PreDestroy;
+
+import com.netflix.titus.common.runtime.TitusRuntime;
+import com.netflix.titus.common.util.rx.ObservableExt;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import rx.Observable;
+import rx.Subscription;
+import rx.subjects.PublishSubject;
+
+public abstract class StreamDataReplicator<D> implements DataReplicator<D> {
```

Comment:
add comment on what is this class and why it is needed.

Reply:
Done